### ISSUE TITLE
Pattern matching on string gives unexpected unreachable clause

### ISSUE THREAD

**Issue by guillaumebrunerie** [agda/agda#2954]
The code below gives the warning `Unreachable clause when checking the definition of test` and highlights the second clause (with `right`), and normalizing `test "a" "x"` gives `nowhere` instead of `right`.
This is surprising, why is the second clause unreachable? Compare for instance to the code in #2952, which is very similar but works as expected.
(tested on the stable branch)

```agda
open import Data.String.Base

test : String → String → String
test "x" b = "left"
test a "x" = "right"
test a b = "nowhere"
```

**Comment by jespercockx** [comment]
This bug was probably introduced by @andreasabel in https://github.com/agda/agda/commit/c688d09960f8e246b2bfc63d81c2ae1b3f34bc5e (fix of #2243).

**Comment by andreasabel** [comment]
See also #708.

**Comment by andreasabel** [comment]
In comparison to #2243, it seems we need to distinguish whether duplicate literals like `"x"` come from the same column (#2243) or different column (this issue).

**Comment by UlfNorell** [comment]
I feel like this issue is a lot worse than #2243, so absent a proper fix we should roll back the fix to #2243 before release.
